src/bpfbytecode.cpp

```diff
@@ -44,6 +53,34 @@ const std::vector<uint8_t> &BpfBytecode::getSection(
   return sections_.at(name);
 }
 
+BpfProgram &BpfBytecode::getProgramForProbe(const Probe &probe)
+{
+  auto usdt_location_idx = (probe.type == ProbeType::usdt)
+                               ? std::make_optional<int>(
+                                     probe.usdt_location_idx)
+                               : std::nullopt;
+
+  auto prog = programs_.find(
+      get_function_name_for_probe(probe.name, probe.index, usdt_location_idx));
```

Comment:
maybe for a follow-up diff but it feels like `get_function_name_for_probe` should be handling the usdt logic

Reply:
Yeah, this is a bit messy but I'd rather refactor it separately.